samples/src/main.c

```diff
@@ -3,27 +3,197 @@
+
+	count = 0;
+}
 
 int main(void)
 {
-	const struct device *const dev = DEVICE_DT_GET_ANY(ilitek_ili9163c);
-	struct sensor_value value;
+	char count_str[11] = {0};
+	const struct device *display_dev;
+	lv_obj_t *hello_world_label;
+	lv_obj_t *count_label;
 
-	if (!device_is_ready(dev)) {
-		printk("Device %s is not ready\n", dev->name);
-		return 1;
+	display_dev = DEVICE_DT_GET(DT_CHOSEN(zephyr_display));
+	if (!device_is_ready(display_dev)) {
+		LOG_ERR("Device not ready, aborting test");
+		return 0;
 	}
 
-	while (1) {
-		sensor_sample_fetch(dev);
+#ifdef CONFIG_GPIO
+	if (gpio_is_ready_dt(&button_gpio)) {
+		int err;
+
+		err = gpio_pin_configure_dt(&button_gpio, GPIO_INPUT);
+		if (err) {
+			LOG_ERR("failed to configure button gpio: %d", err);
+			return 0;
+		}
+
+		gpio_init_callback(&button_callback, button_isr_callback, BIT(button_gpio.pin));
+
+		err = gpio_add_callback(button_gpio.port, &button_callback);
+		if (err) {
+			LOG_ERR("failed to add button callback: %d", err);
+			return 0;
+		}
+
+		err = gpio_pin_interrupt_configure_dt(&button_gpio, GPIO_INT_EDGE_TO_ACTIVE);
+		if (err) {
+			LOG_ERR("failed to enable button callback: %d", err);
+			return 0;
+		}
+	}
+#endif
+
+#ifdef CONFIG_LV_Z_ENCODER_INPUT
+	lv_obj_t *arc;
+	lv_group_t *arc_group;
+
+	arc = lv_arc_create(lv_scr_act());
+	lv_obj_align(arc, LV_ALIGN_CENTER, 0, -15);
+	lv_obj_set_size(arc, 150, 150);
 
-		// Example for ambiant temperature sensor
-		sensor_channel_get(dev, SENSOR_CHAN_AMBIENT_TEMP, &value);
-		printk("Value: %d.%06d\n", value.val1, value.val2);
+	arc_group = lv_group_create();
+	lv_group_add_obj(arc_group, arc);
+	lv_indev_set_group(lvgl_input_get_indev(lvgl_encoder), arc_group);
+#endif
 
-		k_sleep(K_MSEC(1000));
+#ifdef CONFIG_LV_Z_KEYPAD_INPUT
+	lv_obj_t *btn_matrix;
+	lv_group_t *btn_matrix_group;
+	static const char *const btnm_map[] = {"1", "2", "3", "4", ""};
+
+	btn_matrix = lv_btnmatrix_create(lv_scr_act());
+	lv_obj_align(btn_matrix, LV_ALIGN_CENTER, 0, 70);
+	lv_btnmatrix_set_map(btn_matrix, (const char **)btnm_map);
+	lv_obj_set_size(btn_matrix, 100, 50);
+
+	btn_matrix_group = lv_group_create();
+	lv_group_add_obj(btn_matrix_group, btn_matrix);
+	lv_indev_set_group(lvgl_input_get_indev(lvgl_keypad), btn_matrix_group);
+#endif
+
+	if (IS_ENABLED(CONFIG_LV_Z_POINTER_INPUT)) {
+		lv_obj_t *hello_world_button;
+
+		hello_world_button = lv_btn_create(lv_scr_act());
+		lv_obj_align(hello_world_button, LV_ALIGN_CENTER, 0, -15);
+		lv_obj_add_event_cb(hello_world_button, lv_btn_click_callback, LV_EVENT_CLICKED,
+				    NULL);
+		hello_world_label = lv_label_create(hello_world_button);
+	} else {
+		hello_world_label = lv_label_create(lv_scr_act());
+	}
+
+	lv_label_set_text(hello_world_label, "6TRON BY CATIE!");
+	lv_obj_align(hello_world_label, LV_ALIGN_CENTER, 0, 0);
+
+	count_label = lv_label_create(lv_scr_act());
+	lv_obj_align(count_label, LV_ALIGN_BOTTOM_MID, 0, 0);
+
+	lv_task_handler();
+	display_blanking_off(display_dev);
```

Comment:
Rework to use only the ilitek display, without lvgl_input

Reply:
Resolved in [1107a99](https://github.com/catie-aq/zephyr_ilitek-ili9163c/pull/3/commits/1107a9950e66be1f2fa57e347fc6bf2b0cb3b9f8).